src/components/PayloadDetailsCard.js

```diff
@@ -241,37 +260,31 @@ function ExtrinsicPart({ label, fallback, prefix, value }) {
 	};
 
 	const renderMethodDetails = () => {
-		return (
-			argNameValue &&
-			sectionMethod && (
-				<View style={{ display: 'flex', flexDirection: 'column' }}>
-					<Text style={styles.secondaryText}>
-						You are calling{' '}
-						<Text style={styles.secondaryText}>{sectionMethod}</Text> with the
-						following arguments:
-					</Text>
-					{Object.entries(argNameValue).map(([key, argValue]) => {
-						return (
-							<View
-								key={key}
-								style={{
-									alignItems: 'flex-start',
-									display: 'flex',
-									flexDirection: 'row',
-									flexWrap: 'wrap',
-									padding: 5
-								}}
-							>
-								<Text style={{ ...styles.subLabel, flex: 1 }}>{key}: </Text>
-								<Text style={{ ...styles.secondaryText, flex: 3 }}>
-									{argValue}
-								</Text>
-							</View>
-						);
-					})}
-				</View>
-			)
-		);
+		if (formattedCallArgs) {
+			return Object.entries(formattedCallArgs)
+				.reverse()
+				.map((entry, index) => {
+					const sectionMethod = entry[0];
+					const paramArgs = entry[1];
+
+					return (
+						<View key={index} style={styles.callDetails}>
+							<Text>
+								Call <Text style={styles.secondaryText}>{sectionMethod}</Text>{' '}
+								with the following arguments:
+							</Text>
+							{paramArgs.map(([param, arg]) => (
+								<React.Fragment key={param}>
+									<Text style={{ ...styles.subLabel, flex: 1 }}>{param}: </Text>
```

Comment:
`flex` on  your iOS device also does not work, on Android the view with `flex` will disappear. I think it is because with `flex-direction: column` we have to set the `height` otherwise the `flex` will not work. 

Consider the height might be variant since:
1. nested call with "sudo"
2. the args could be super long, e.g. args for "setCode"

Probably we just do not set height, but gives some styling, and use ellipses for super long args.